### PR TITLE
Use Guzzle RequestOptions constants for configuring requests

### PR DESCRIPTION
**Pull request by deguif**
Use RequestOptions constants.
The main advantage I see is that it will be easier to spot removed options ;)

### CODE REVIEW

File: src/Transport/Guzzle.php [hunk 1]
```diff
@@ -57,13 +58,14 @@ public function exec(Request $request, array $params): Response
 
         $options = [
             'base_uri' => $this->_getBaseUrl($connection),
-            'headers' => [
+            RequestOptions::HEADERS => [
                 'Content-Type' => $request->getContentType(),
             ],
             'exceptions' => false, // 4xx and 5xx is expected and NOT an exceptions in this context
```
Comment (deguif) [inline src/Transport/Guzzle.php:64]: Will be done for this option too when https://github.com/ruflin/Elastica/pull/1817 will be merged.

**Review by ruflin — APPROVED** [review]
LGTM. Conflict on the changelog which needs resolving.

**Comment by deguif** [comment]
Just rebased ;)

**Review by ruflin — APPROVED** [review]
LGTM, please merge when green.

**Pull request merged**